zerver/actions/message_flags.py

```diff
@@ -229,18 +229,28 @@ def do_clear_mobile_push_notifications_for_ids(
     # only for the message-edit use case where we'll have a single message_id.
     assert len(user_profile_ids) == 1 or len(message_ids) == 1
 
-    messages_by_user = defaultdict(list)
+    push_device_registered_user_ids = (
+        PushDeviceToken.objects.filter(user_id__in=user_profile_ids)
+        .values_list("user_id", flat=True)
+        .union(
+            # Uses index "zerver_pushdevice_user_bouncer_device_id_idx".
+            PushDevice.objects.filter(
+                user_id__in=user_profile_ids, bouncer_device_id__isnull=False
+            ).values_list("user_id", flat=True)
+        )
+    )
     notifications_to_update = (
         UserMessage.objects.filter(
             message_id__in=message_ids,
-            user_profile_id__in=user_profile_ids,
+            user_profile_id__in=push_device_registered_user_ids,
```

Comment:
I don't think this is the right layer to do this filtering at. We do want to clear `where_active_push_notification` in our database when the user has no registered devices, whereas this will leak that flag on those users forever?